envoy/filesystem/filesystem.h

```diff
@@ -169,6 +169,9 @@ struct DirectoryEntry {
   // target. For example, if name_ is a symlink to a directory, its file type will be Directory.
   FileType type_;
 
+  // The file size in bytes.
+  uint64_t size_bytes_;
```

Comment:
should we include size_bytes_ in the operator== impl? Actually it is not obvious why you previously needed one of those. Would the default operator not work?

WIth this explicit definition, now it will be different of course as this one ignores the size.

Also, what is this operator== used for?

Reply:
Yikes, good catch. The == is at least used by the test, so fixing it revealed that I wasn't testing what I thought I was testing. It also seems to be used in a != operator for the iterator, for the `!= end()` comparison, which didn't really care about the size part. Fixed.

Reply:
Oh, and there is no default == operator apparently. I too thought there was for pod types, but apparently not until C++20 and even then you still have to specify it with `=default`. I guess we've both been mixing languages too much!

Reply:
I *think* keeping the `==` (but fixed) is the least-confusing option for this. It's unfortunate that iterators use `!=` as the 'is at the end' check. We *could* override that to be specifically a test for end-iterator but that would be way more confusing than having an otherwise-unnecessary == operator.

And having the == operator be only as specific as it needs to be for iterator end detection would also just be more confusing than helpful, and would require rewriting the tests significantly. So I think what we have now (a handwritten default == operator) is the least horrible option.

Reply:
sounds good